modules/phase_field/src/materials/AsymmetricCrossTermBarrierFunctionMaterial.C

```diff
@@ -0,0 +1,97 @@
+      const Real nj = (*_eta[j])[_qp];
+
+      const Real Wij = _W_ij[_num_eta * i + j];
+      const Real Wji = _W_ij[_num_eta * j + i];
+
+      switch (_g_order)
+      {
+        case 0: // SIMPLE
+          _prop_g[_qp] += 16.0 * (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * (ni * ni * nj * nj);
+          // first derivatives
+          (*_prop_dg[i])[_qp] += 16.0 * (  (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * (2 * ni * nj * nj)
+                                         + (Wij * (*_dh[i])[_qp]) * (ni * ni * nj * nj));
+          (*_prop_dg[j])[_qp] += 16.0 * (  (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * (2 * ni * ni * nj)
+                                         + (Wji * (*_dh[j])[_qp]) * (ni * ni * nj * nj));
+          // second derivatives (diagonal)
+          (*_prop_d2g[i][i])[_qp] += 16.0 * (  (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * (2 * nj * nj)
+                                             + 2 * (Wij * (*_dh[i])[_qp]) * (2 * ni * nj * nj)
+                                             + (Wij * (*_d2h[i])[_qp]) * (ni * ni * nj * nj));
+          (*_prop_d2g[j][j])[_qp] += 16.0 * (  (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * (2 * ni * ni)
+                                             + 2 * (Wji * (*_dh[j])[_qp]) * (2 * ni * ni * nj)
+                                             + (Wji * (*_d2h[j])[_qp]) * (ni * ni * nj * nj));
+          // second derivatives (off-diagonal)
+          (*_prop_d2g[i][j])[_qp] = 16.0 * (  (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * (4 * ni * nj)
+                                             + (Wji * (*_dh[j])[_qp]) * (2 * ni * nj * nj)
+                                             + (Wij * (*_dh[i])[_qp]) * (2 * ni * ni * nj));
+          break;
+
+        case 1: // LOW
+          _prop_g[_qp] += 4.0 * (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * (ni * nj);
+          // first derivatives
+          (*_prop_dg[i])[_qp] += 4.0 * (  (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * nj
+                                         + (Wij * (*_dh[i])[_qp]) * (ni * nj));
+          (*_prop_dg[j])[_qp] += 4.0 * (  (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp]) * ni
+                                         + (Wji * (*_dh[j])[_qp]) * (ni * nj));
+          // second derivatives (diagonal)
+          (*_prop_d2g[i][i])[_qp] += 4.0 * (  2 * (Wij * (*_dh[i])[_qp]) * nj
+                                            + (Wij * (*_d2h[i])[_qp]) * ni * nj);
+          (*_prop_d2g[j][j])[_qp] += 4.0 * (  2 * (Wji * (*_dh[j])[_qp]) * ni
+                                            + (Wji * (*_d2h[j])[_qp]) * ni * nj);
+          // second derivatives (off-diagonal)
+          (*_prop_d2g[i][j])[_qp] = 4.0 * (  (Wij * (*_h[i])[_qp] + Wji * (*_h[j])[_qp])
+                                            + (Wji * (*_dh[j])[_qp]) * nj
+                                            + (Wij * (*_dh[i])[_qp]) * ni);
+          break;
+
+        default:
+          mooseError("Internal error");
```

Comment:
A user should not be able to trigger this error ever, because the MooseEnum input values are checked at construction time. This is, however, standard practice (at least in the PF module) because an accidental overwrite of the internal state of the enum (due to an unrelated bug) could cause uninitialized values. Also a static code analysis might complain about uninitialized values.